lib/Sema/TypeChecker.cpp

```diff
@@ -626,6 +626,12 @@ void swift::performTypeChecking(SourceFile &SF, TopLevelContext &TLC,
                                 unsigned ExpressionTimeoutThreshold) {
   if (SF.ASTStage == SourceFile::TypeChecked)
     return;
+  
+  // Use a recursive timer in case this func is recursive.
```

Comment:
It shouldn't be. Is this coming up in the test suite?

Reply:
No, I'll take it out, thanks.